### PR TITLE
Add accessibility testing tools

### PR DESCRIPTION
**Pull request by aednlaxer**
- [x] New example app with code comments for some issues
- [x] Testing tools panel
- [x] Option to control logging level (verbose, warnings, none)
- [x] README updates
- [x] Verify changes with the designer
- [x] Fix design notes

### CODE REVIEW

**Review by utopicnarwhal — CHANGES_REQUESTED** [review]

File: example/lib/app_localizations.dart [hunk 1]
```diff
@@ -0,0 +1,184 @@
+import 'package:flutter/material.dart';
+
+class AppLocalizations {
```
[the remark this answers is not in the document]
Comment (aednlaxer) [reply example/lib/app_localizations.dart:16]: It's created manually to let users play with localizations toggle in the example app. I don't think we need to care much about placing localizations in ARB files

File: example/lib/main.dart [hunk 2]
```diff
@@ -10,15 +14,43 @@ class MyApp extends StatelessWidget {
...
       builder: (context, child) {
+        // Add AccessibilityTools to the widget tree. The tools are available
+        // only in debug mode
         return AccessibilityTools(
           checkFontOverflows: true,
           child: child,
         );
       },
-      theme: ThemeData(primarySwatch: Colors.blue),
+      theme: ThemeData(
```
[the remark this answers is not in the document]
Comment (aednlaxer) [reply example/lib/main.dart]: I've added a dark theme. I'd like not to add high contrast themes just now. In the future I'd like to have a toggle that changes the high contrast flag, let's add themes then

File: lib/src/accessibility_tools.dart [hunk 3]
```diff
@@ -48,7 +68,13 @@ class AccessibilityTools extends StatefulWidget {
 
 class _AccessibilityToolsState extends State<AccessibilityTools>
     with SemanticUpdateMixin {
-  late CheckerManager _checker = CheckerManager(_getCheckers());
+  late CheckerManager _checker = CheckerManager(
```
[the remark this answers is not in the document]
Comment (aednlaxer) [reply lib/src/accessibility_tools.dart:71]: It's just a way to initalize something not `final` later without using the `initState` method

File: lib/src/accessibility_tools.dart [hunk 4]
```diff
@@ -118,12 +147,40 @@ class _AccessibilityToolsState extends State<AccessibilityTools>
     return Stack(
```
[the remark this answers is not in the document]
Comment (aednlaxer) [reply lib/src/accessibility_tools.dart:147]: I've played with this and decided not to add it:
1. I'm okay with testing tools using main app's theme even if it's not using M3
2. I'd like to follow user's selected theme (dark or light) for _accessibility_ purposes :) 
 
Adding 
```dart
data: ThemeData(useMaterial3: true),
```
seems to break light-dark theme switching. Creating monsters like

```dart
data: ThemeData.from(
    colorScheme: Theme.of(context).colorScheme,
    useMaterial3: true,
),
```
doesn't seem right

**Pull request merged**